modules/config/home-cursor.nix

```
@@ -142,6 +146,11 @@ in {
         XCURSOR_SIZE = mkDefault cfg.size;
         XCURSOR_THEME = mkDefault cfg.name;
       };
+
+      # Symlink $HOME/.icon to the home-manager profile icons directory, needed for
+      # backwards compatibility with some applications. See:
+      # https://specifications.freedesktop.org/icon-theme-spec/latest/ar01s03.html
+      home.file.".icons".source = "${config.home.profileDirectory}/share/icons";
```

Unfortunately this does not work in pure evaluation mode:

> error: access to absolute path '/home/rycee/.nix-profile/share/icons' is forbidden in pure eval mode (use '--impure' to override)

Maybe

```suggestion
      home.file.".icons".source = "${config.home.path}/share/icons";
```

instead?

---

A concern here may be that users may wish to place their own icons non-declaratively inside `~/.icons` and this would prevent that. Perhaps `recursive = true` is a possibility but it may be a bit slow if you have many icons installed…

---

Given the scope of the home-pointecursor module pertains to cursor configurations, I opted to symlink only the leaf files from `cfg.package` rather than the entire icons directory to mitigate the performance impact incurred by `recursive = true`. This leaves somewhat more to be desired but I think a clear separation of concerns is appropriate here. On a technicality, the user can still pass a symlink farm to `cfg.package` and potentially experience slowdowns due to link creation, however I think in that case it can assumed that the user is able to wrangle and fine tune the directory structure manually to their needs.1